internal/services/eventgrid/eventgrid_partner_configuration_resource.go

```diff
@@ -0,0 +1,270 @@
+	}
+}
+
+func (EventGridPartnerConfigurationResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (EventGridPartnerConfigurationResource) ModelObject() interface{} {
+	return &EventGridPartnerConfigurationResourceModel{}
+}
+
+func (EventGridPartnerConfigurationResource) ResourceType() string {
+	return "azurerm_eventgrid_partner_configuration"
+}
+
+func (r EventGridPartnerConfigurationResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.PartnerConfigurations
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var config EventGridPartnerConfigurationResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			// Uses resource group ID as it has 1-1 mapping with resource group
+			// See the SDK sample usage: https://github.com/hashicorp/go-azure-sdk/tree/main/resource-manager/eventgrid/2022-06-15/partnerconfigurations
+			id := commonids.NewResourceGroupID(subscriptionId, config.ResourceGroup)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			param := partnerconfigurations.PartnerConfiguration{
+				Location: pointer.To("global"),
+				Properties: &partnerconfigurations.PartnerConfigurationProperties{
+					PartnerAuthorization: &partnerconfigurations.PartnerAuthorization{
+						DefaultMaximumExpirationTimeInDays: pointer.To(config.DefaultMaximumExpirationTimeInDays),
+						AuthorizedPartnersList:             expandAuthorizedPartnersList(&config.PartnerAuthorizations),
+					},
+				},
+				Tags: pointer.To(config.Tags),
+			}
+			if err := client.CreateOrUpdateThenPoll(ctx, id, param); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r EventGridPartnerConfigurationResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.EventGrid.PartnerConfigurations
+
+			id, err := commonids.ParseResourceGroupID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var config EventGridPartnerConfigurationResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			if existing.Model == nil {
+				return fmt.Errorf("retrieving %s: `model` was nil", *id)
+			}
+
```

Comment:
```suggestion
			if existing.Model.Properties == nil {
				return fmt.Errorf("retrieving %s: `model.Properties` was nil", *id)
			}
			payload := existing.Model
			if payload.Properties.PartnerAuthorization == nil {
				payload.Properties.PartnerAuthorization = &partnerconfigurations.PartnerAuthorization{}
			}
```

Reply:
Then replace instances of `existing.Model` with `payload` to conform more closely to the contribution guide.